pysal/network/network.py

```diff
@@ -627,19 +632,50 @@ def enum_links_node(self, v0):
             links.append(tuple(sorted([n, v0])))
         return links
 
-    def node_distance_matrix(self):
+    def node_distance_matrix(self, processing):
+        """
+        Called from: allneighbordistances()
+                     nearestneighbordistances()
+                     distancebandweights()
+        
+        """
         self.alldistances = {}
         nnodes = len(self.node_list)
         self.distancematrix = np.empty((nnodes, nnodes))
-        for node in self.node_list:
-            distance, pred = util.dijkstra(self, self.edge_lengths, node, n=float('inf'))
+        # Single-core processing
+        if not processing:
+            for node in self.node_list:
+                distance, pred = util.dijkstra(self, self.edge_lengths, node, n=float('inf'))
+                pred = np.array(pred)
+                #tree = util.generatetree(pred)     <---- something to look at in the future
+                tree = None
+                self.alldistances[node] = (distance, tree)
+                self.distancematrix[node] = distance
+        # Multiprocessing
+        if processing:
+            try:
+                import multiprocessing as mp
+                from itertools import repeat
+            except:
+                raise Execption("Must have multiprocessing and itertools")
+            if processing == 9999:
```

Comment:
I think it makes more sense to make this a separate type of argument, like `'all'` or `np.inf`, rather than a large int.